src/composables/use-event-listener.js

```diff
@@ -0,0 +1,33 @@
+import {
+  isRef,
+  watch,
+  onMounted,
+  onBeforeUnmount,
+  unref,
+} from '@nuxtjs/composition-api'
+
+/**
+ * Use an event listener. Shamelessly stolen from https://logaretm.com/blog/my-favorite-5-vuejs-composables/#useeventlistener
+ *
+ * @param {Ref<EventTarget | null> | EventTarget} target The target can be a reactive ref which adds flexibility
+ * @param {string} event
+ * @param {(e: Event) => void} handler
+ */
+export function useEventListener(target, event, handler) {
+  // if its a reactive ref, use a watcher
+  if (isRef(target)) {
+    watch(target, (value, oldValue) => {
+      oldValue?.removeEventListener(event, handler)
+      value?.addEventListener(event, handler)
+    })
```

Comment:
Would it be better to use `onInvalidate` to remove the event listener?

```suggestion
    watch(target, (value, _, onInvalidate) => {
      value?.addEventListener(event, handler)
      onInvalidate(() => value?.removeEventListener(event, handler))
    })
```

I'm not 100% sure there is a difference, I'll do some fiddling to test this out, but i think the watcher may never run on component dismount to remove the old listener but invalidate will run.

Reply:
I do like your take on it! 😄